Libplanet/Net/Swarm.TrustedStateCompleter.cs

```diff
@@ -0,0 +1,147 @@
+            {
+                var request = new GetBlockStates(blockHash);
+                foreach (BoundPeer peer in trustedPeers)
+                {
+                    _logger.Debug(
+                        "Requests the block {BlockHash}'s states to a trusted peer {Peer}...",
+                        blockHash,
+                        peer
+                    );
+                    Message reply;
+                    try
+                    {
+                        reply = await Transport.SendMessageWithReplyAsync(
+                            peer,
+                            request,
+                            timeout: Options.RecentStateRecvTimeout,
+                            cancellationToken: cancellationToken
+                        );
+                    }
+                    catch (TimeoutException e)
+                    {
+                        string msg = "Failed to receive the block {BlockHash}'s states from " +
+                            "a trusted peer {Peer}: " + e;
+                        _logger.Error(e, msg, blockHash, peer);
+                        continue;
+                    }
+
+                    if (reply is BlockStates blockStates)
+                    {
+                        if (!blockStates.BlockHash.Equals(blockHash))
+                        {
+                            string msg =
+                                $"Expected a {nameof(BlockStates)} message for the block " +
+                                "{RequestedBlockHash}, but received a wrong reply instead from " +
+                                "a trusted peer {Peer}: {RepliedBlockHash}";
+                            _logger.Error(msg, blockHash, peer, blockStates.BlockHash);
+                            continue;
+                        }
+                        else if (blockStates.States is null)
+                        {
+                            string msg =
+                                "A trusted peer {Peer} does not have the block {BlockHash}'s " +
+                                "states.";
+                            _logger.Error(msg, peer, blockHash);
+                            continue;
+                        }
+
+                        BlockChain.Store.SetBlockStates(blockStates.BlockHash, blockStates.States);
+                    }
+                    else
+                    {
+                        string msg =
+                            $"Expected a {nameof(BlockStates)} message, but received a wrong " +
+                            "reply instead from a trusted peer {Peer}: {ReplyMessage}";
+                        _logger.Error(msg, peer, reply);
+                    }
+                }
+            }
+
+            return new StateCompleterSet<T>
+            {
+                StateCompleter = (blockChain, blockHash, address) =>
+                {
+                    FillTrustedBlockStates(blockHash).Wait(cancellationToken);
```

Comment:
It can be a matter if `StateCompleter` is called in `async` method... but it doesn't seem translatable to `async` because `StateCompleter` and related features are all synchronous. 🤔

Reply:
That's what I'd grappled with, but I gave up and decided to request for comments about this problem…

Reply:
@longfin I'm going to address this problem in the next separated pull request!